pkg/ebpf/tracer.go

```diff
@@ -387,17 +388,26 @@ func (t *Tracer) removeEntries(mp, tcpMp *bpflib.Map, entries []*ConnTuple) {
 }
 
 // getTCPStats reads tcp related stats for the given ConnTuple
-func (t *Tracer) getTCPStats(mp *bpflib.Map, tuple *ConnTuple) *TCPStats {
+func (t *Tracer) getTCPStats(mp *bpflib.Map, tuple *ConnTuple, seen map[ConnTuple]struct{}) *TCPStats {
 	// The PID isn't used as a key in the stats map, we will temporarily set it to 0 here and reset it when we're done
 	pid := tuple.pid
 	tuple.pid = 0
 
 	stats := &TCPStats{retransmits: 0}
 
+	// Since we can't reliably use the PID for tracking TCPStats, we only report retransmits for
```

Comment:
its a nit but can we return early if the connection is not a tcp connection ?

Reply:
👍 will update it